src/opm/input/eclipse/EclipseState/Grid/NNC.cpp

```diff
@@ -171,14 +177,81 @@ bool is_neighbor(const EclipseGrid& grid, std::size_t g1, std::size_t g2) {
         }
     }
 
+    void NNC::load_editr(const EclipseGrid& grid, const Deck& deck) {
+        // Will contain EDITNNCR data in reverse order to be able
+        // use unique to only keep the last one from the data file.
+        std::deque<NNCdata> nnc_editr;
+
+        const auto& keyword_list = deck.getKeywordList<ParserKeywords::EDITNNCR>();
+
+        if (keyword_list.empty()) {
+            return;
+        }
+
+        for (const auto& keyword_ptr : deck.getKeywordList<ParserKeywords::EDITNNCR>()) {
+            const auto& records = *keyword_ptr;
+            if (records.empty()) {
+                continue;
+            }
+            for (const auto& record : records) {
+                auto index_pair = make_index_pair(grid, record);
+                if (!index_pair)
+                    continue;
+
+                auto [g1, g2] = index_pair.value();
+                if (is_neighbor(grid, g1, g2))
+                    continue;
+
+                double trans = record.getItem(6).getSIDouble(0);
+                nnc_editr.emplace_front(g1, g2, trans);
+            }
+
+            if (!this->m_editr_location)
+                this->m_editr_location = keyword_ptr->location();
+        }
+
+        if (nnc_editr.empty()) {
+            return;
+        }
+
+        // Make the entries unique (i.e. only one entry for each index pair)
+        // the entry surviving should be last one (i.e. later enrtries overwrite previous entries)
+        // Will keep the insertion order of entries for same cell pair.
```

Comment:
I don't understand this comment.  Are you stating/assuming that `std::unique()` preserves the **last** item in a group of equivalent items?  If so, that's not guaranteed by the standard.  In fact, the standard pretty much guarantees the opposite&mdash;i.e., that `std::unique()` preserves the **first** item in a group of equivalent items.

If you're talking about something else then the comment should be clarified.

Reply:
I think the logic here is subtly wrong, but not much: I think using std::stable_sort() below is sufficient. Without that, I do not believe it is guaranteed that you preserve the order of elements.

With that change, I think the comment above is fine as std::unique will keep the last entry, but they were inserted in reverse order.

Reply:
I admit I did not read the full algorithm, but I stand by my original remark.  The comment is (potentially) misleading, because it **appears to** be talking about `std::unique()` in isolation and does not mention the essential `emplace_front()` operation.  I'd prefer if you'd at least add that to the text.

As for `sort()`/`unique()`, @atgeirr is correct.  Since you ultimately depend on the relative order of the items&mdash;i.e., last element entered should be first in sorted group&mdash;you need to use `stable_sort()` here.  Regular `sort()` can and will permute the order of equivalent elements.

Reply:
I tried to make the comments in the code more clear. Open for further suggestions.